python/cudf/cudf/_lib/transpose.pyx

```diff
@@ -64,7 +64,7 @@ def transpose(Table source):
         result = Table(index=result._index, data=[
             (name, cudf.core.column.column.build_categorical_column(
                 codes=cudf.core.column.column.as_column(
-                    col.base_data, dtype=col.dtype),
+                    col.data, dtype=col.dtype),
```

Comment:
Instead of passing `col.data` here we should pass `col.base_data` and `col.offset` as the offset